noisy-simulator/src/operation/tests.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use nalgebra::dmatrix;
+use num_complex::{Complex, ComplexFloat};
+
+use crate::{
+    operation::{operation, Operation},
+    tests::assert_approx_eq,
+    SquareMatrix,
+};
+
+/// Check that the inner matrices of the instrument are constructed correctly.
+#[test]
+fn constructor() {
```

Comment:
What in our code is being tested here?

Reply:
That the inner matrices of an instrument are constructed correctly. Namely, the "effect" matrix and the "operation" matrix.

Reply:
Should the test just do that then? It looks like is it doing a lot more than verifying that a macro constructs a matrix properly.